### DISCUSSION TITLE
Allow cameras listed in a camera group to have non-16:9 placements

### DISCUSSION THREAD

**Discussion by autumntxx** [blakeblackshear/frigate#13582] [Ideas]
I may be doing something wrong, but when running Frigate 0.14.1, camera groups enforce cameras to be a seemingly 16:9 aspect ratio. For example, here's the gaps I end up with when using 1280x960 cameras:
![image](https://github.com/user-attachments/assets/51c6ca3c-1fbc-4517-acaf-0c5385d9973d)

Maybe a quick hotkey could be added, like holding down shift while resizing, and that would allow you to resize their placements to any aspect ratio. From there, the video feeds would scale to be as big as possible inside of those placements, instead of just scaling to their biggest vertically.

This might also already be added and I'm just missing something. Anyways, love the development so far!

**Comment by hawkeye217** [comment]
This is expected and by design, see the [FAQ](https://github.com/blakeblackshear/frigate/discussions/12724), Usage question 4.

It's not mathematically possible to retain a grid with randomly sized aspect ratio boxes. There will always be wasted/black space.

**Comment by autumntxx** [comment]
I apologize, I didn't see that section of the FAQ.

However, wouldn't it still be technically possible to add? Here's what I was imagining, demonstrated in Grafana:
![firefox_vMCWWbjZ2I](https://github.com/user-attachments/assets/7701e793-2850-46a3-a917-66fd9a4e9342)

While it wouldn't completely remove borders, I feel like it'd make it a lot better if you only have 1280x960 cameras

**Reply by hawkeye217** [reply]
That isn't really all that different than what exists now. Technically, more columns could be added to the grid to reduce the amount of empty space, but you're still going to have the same issues, especially if you begin mixing cameras with more unique aspect ratios (wide-angle, dual lens, portrait doorbell, etc).

**Reply by ProsjektX** [reply]
Why not have an option for layout without grid? Just masonry style. E.g. with portrait cameras in between there will be much space
![image](https://github.com/user-attachments/assets/9c77990c-49a1-401c-b49f-70bbf1d5f2a1)

**Reply by NickM-27** [reply]
Because in a masonry layout you can't have cells that take up multiple horizontal or vertical slots which is important for panoramic and portrait cameras
